### ISSUE TITLE
The handling of renamed modules should be improved

### ISSUE THREAD

**Issue by mietek** [agda/agda#2092]
Let’s say we are reorganising a repository, and we rename the file “Foo.agda” to “Bar.agda”, in the filesystem. Next, we update “Everything.agda” to `import Bar`, and we reload the file.  We get an error:

> The name of the top level module does not match the file name. The module Bar should be defined in one of the following files: …

First, the error message is not very helpful.  This could be improved to say “The file `Bar.agda` should contain the `Bar` top level module, but it contains the `Foo` top level module.”, or something to this effect.

Second, the error leaves Emacs in a black-and-white state, pointed at the import site in “Everything.agda”, which means it’s not possible to easily navigate to the problematic file, that is, “Bar.agda”.  This could be improved by pointing Emacs at the problematic `Foo` top level module declaration, in the problematic file.

**Comment by andreasabel** [comment]
1. I think the error message is so complicated because of hierarchical modules.  Of course, Agda could suggest the most likely fix.
2. The other concern is that the error is raised in the wrong location.

I addressed 2. by raising the error in the file with the wrong module name.

Wrt. 1., what currently Agda does it finds the file "Bar.agda" for the import of "Bar".  Then it processes this file without considering how it got there. It realizes that the module name does not match a valid filepath (relative to the current include directories) and complains.  We can distinguish whether a file load came from the top or via an import, and give different error messages for these cases.

**Comment by andreasabel** [comment]
Ok, now I addressed also 1.  Required some digging through my (least) favorite Spaghetti code, which got some small clarification along the way: `Agda.Interaction.Imports`.
Now, Agda will suggest you name the wrongly named module as you said so in the import statement.

**Comment by nad** [comment]
> I addressed 2. by raising the error in the file with the wrong module name.

The (history of the) error message for `Issue2092.agda` suggests otherwise. It starts in the following way:
```
Issue2092.agda:1,8-34
```
Did you change your mind?

(If anyone decides to do something about this, please wait until after I have pushed a fix for #2473.)

**Comment by nad** [comment]
> (If anyone decides to do something about this, please wait until after I have pushed a fix for #2473.)

Done.

**Comment by mietek** [comment]
Is there any chance to get this out of the icebox?

**Comment by jespercockx** [comment]
Yes of course there is a chance, all you need to do is to start working on it or convince someone else to do so :)

**Comment by mietek** [comment]
I was hoping that @andreasabel could say what happened to his solution to (2).

> I addressed 2. by raising the error in the file with the wrong module name.

**Comment by mietek** [comment]
There is a viable workaround: Load an `Everything.agda` directory file that points to all the files that need to be renamed, then use it to navigate to each file and rename its main module _without reloading any of the files._  Once that's finished, move and/or rename all the files, and only then, reload `Everything.agda`.
